### DISCUSSION TITLE
Setting a favorite layout for each app

### DISCUSSION THREAD

**Discussion by escapist** [rxhanson/RectanglePro-Community#25] [Ideas]
I absolutely love Better Window Manager even though it's a bit buggy. It supports basic app-specific positioning and lets me use the same keyboard shortcut for any of those app presets. I'll _try_ to explain…

I chose a favorite size and position for most apps and assigned it ⌥~ as the keyboard shortcut. This shortcut works on any window _UNLESS_ I've overridden it for a specific app. So, for instance, if I use that shortcut while Things is focused, it moves Things into the size and position I specified for _that app_. If Fantastical is focused, I can use the same shortcut to move it into the position I chose for _that app_. And if I haven't specified an override for a given app, it falls back to the size and position I specified for _all apps_ when ⌥~ is pressed.

This is tough to describe. I hope it made sense.

All this is to say, I'd love to be able to set, say, ⌥~ to move any focused app into a certain position, and to be able to override that position for certain apps using the same keyboard command.

**Comment by rxhanson** [comment]
To achieve this in Rectangle Pro, 

1. Create an app group (or save layout from the menu bar menu).
2. Check the box for "Only execute on frontmost" & give it a keyboard shortcut
3. Add any more of your apps with the positions you want, or update the existing ones.
4. Add the "Global" app and give that the size/position you want for everything else

Let me know if I misunderstood or if that's not clear.

Side note: Currently, saving the apps' existing windows will be in absolute dimensions, and you'll have to translate that over by hand to relative dimensions if that's what you desire (values less than 1 & a docking point). I have plans to make it easier to get relative dimensions in a future update.

**Comment by escapist** [comment]
I think the difference is that I can't use the same keyboard shortcut for
multiple groups, right? So I can't use ^⌥⌘` For Superhuman because I used
it on Things? So I'd have to memorize a bunch of different keyboard
shortcuts?

**Comment by escapist** [comment]
Ooooh wait, I think I understand. So every app I want to control with the
same keyboard shortcut would just go into the same group, and I'd turn on
the feature to only execute on frontmost. I'll try it out.
